File: python-legacy-build/magic/magic_settings.py
import os
import json

class Settings:
    def __init__(self):
        self.level = 99
        self.potion = 'None'
        self.prayer = 'None'
        
       #weapons
        self.mh = {
            'name': 'Wand of the praesul',
            'aftershock': 0,
            'precise': 0,
            'equilibrium': 0,
            'ruthless': 0,
            'spendthrift': 0,
            'flanking': 0
        }
        
        self.oh = {
            'name': 'Imperium core',
            'aftershock': 0,
            'precise': 0,
            'equilibrium': 0,
            'ruthless': 0,
            'spendthrift': 0,
            'flanking': 0
        }
        
        self.th = {
            'name': 'Fractured staff of Armadyl',
            'aftershock': 0,
            'precise': 6,
            'equilibrium': 2,
            'ruthless': 0,
            'spendthrift': 0,
            'flanking': 0
        }
        
        self.sh = {
            'tier': 90,
            'type': 'defender',
            'aftershock': 0,
            'precise': 0,
            'equilibrium': 0,
            'ruthless': 0,
            'spendthrift': 0,
            'flanking': 0
        }
        
        self.helm = 'None'
        self.neck = 'None'
        self.body = 'Elite tectonic'
        self.legs = 'None'
        self.boots = 'None'
        self.gloves = 'None'
        self.ring = 'Reavers'
        self.cape = 'None'
        self.pocket = 'Grimoire'
        self.aura = 'None'
        
        if self.ring == 'None':
            self.reavers = 1
        else:
            self.reavers = 0
        
        if self.pocket == 'Grimoire':
            self.grim = 1
        else:
            self.grim = 0
        
        self.bitingRank = 4
        self.armourLvl = 20  
        
        if self.armourLvl == 20:
            self.biting = self.bitingRank * 0.022
        else:
            self.biting = self.bitingRank * 0.02
         
        self.vuln = 0
        self.sc = 1
        self.fsoa = 0
        self.kalg = 1
        self.kalgSpec = 1
        self.ful = 0
        self.zerkEss = 0
        self.reaperCrew = 1
        self.gconc = 0
        self.enchMeta = 1
        self.enchAff = 1
        self.enchFlame = 1
        self.hex = 0
        self.ripper = 0
        self.dharok = 0
        self.exsang = 0
        self.rubyAurora = 0
        self.slayerSigil = 0
        self.cryptbloom = 0
        self.slayerPerk = 0
        self.nope = 0
        self.kww = 0
        self.revenge = 0
        
        self.utils = Utils()
        
        self.thTier = self.getTier(self.th['name'])
        self.mhTier = self.getTier(self.mh['name'])
        self.ohTier = self.getTier(self.oh['name'])
        self.shTier = self.sh['tier']
        
    def getTier(self, weapon):
        for w in self.utils.weapons:
            if w['name'] == weapon:
                return w['dmg_tier']
        
    def setInput(self, name, value):
        setattr(self, name, value)

class Utils:
    def __init__(self):
        self.gear = self.loadJson('utils', 'gear.json')
        self.weapons = self.loadJson('utils', 'weapons.json')
        self.boosts = self.loadJson('utils', 'boosts.json')
        
    def loadJson(self, directory, filename):
        filePath = os.path.join(directory, filename)
        with open(filePath, 'r') as file:
            return json.load(file)
        
SET_INS = Settings()

File: magic/magic_settings.js
const fs = require('fs');

class Settings {
  constructor(settings) {
    this.level = settings['level'];
    this.potion = settings['potion'];
    this.prayer = settings['prayer'];

    this.mh = settings['mh']

    this.oh = settings['oh']

    this.th = settings['th']

    this.sh = settings['sh']

    this.helm = settings['helm'];
    this.neck = settings['neck'];
    this.body = settings['body'];
    this.legs = settings['legs'];
    this.boots = settings['boots'];
    this.gloves = settings['gloves'];
    this.ring = settings['ring'];
    this.cape = settings['cape'];
    this.pocket = settings['pocket'];
    this.aura = settings['aura'];

    if (this.ring === 'Reavers') {
      this.reavers = 1;
    } else {
      this.reavers = 0;
    }

    if (this.pocket === 'Grimoire') {
      this.grim = 1;
    } else {
      this.grim = 0;
    }

    this.bitingRank = settings['biting-rank'];
    this.armourLvl = True;
    if (this.armourLvl === True){
      this.biting = this.bitingRank * 0.022;
    } else{
      this.biting = this.bitingRank * 0.02;
    }

    this.nope = settings['nope'];
    this.vuln = settings['vuln'];
    this.sc = settings['sc'];
    this.fsoa = settings['fsoa'];
    this.kalg = settings['kalg'];
    this.kalgSpec = settings['kalg-spec'];
    this.ful = settings['ful'];
    this.zerkEss = settings['zerk-ess'];
    this.reaperCrew = settings['reaper-crew'];
    this.gconc = settings['gconc'];
    this.enchMeta = settings['ench-meta'];
    this.enchAff = settings['ench-aff'];
    this.enchFlame = settings['ench-flame'];
    this.hex = settings['hex-hunter'];
    this.ripper = settings['ripper-passive'];
    this.dharok = settings['dharok'];
    this.exsang = settings['exsang'];
    this.rubyAurora = settings['ruby-aurora'];
    this.slayerSigil = settings['sigil'];
    this.cryptbloom = settings['cryptbloom'];
    this.slayerPerk = settings['slayer-perk'];
    this.kww = settings['kww'];

    this.utils = new Utils();

    this.thTier = this.getTier(this.th.name);
    this.mhTier = this.getTier(this.mh.name);
    this.ohTier = this.getTier(this.oh.name);
    this.shTier = this.sh.tier;
  }

  getTier(weapon) {
    for (const w of this.utils.weapons) {
      if (w.name === weapon) {
        return w.dmg_tier;
      }
    }
  }
}

class Utils {
  constructor() {
    this.gear = require('../utils/gear.json');
    this.weapons = require('../utils/weapons.json');
    this.boosts = require('../utils/boosts.json');
  }
}

module.exports = Settings;